data/scripts.js

```diff
@@ -614,15 +623,15 @@ let BattleScripts = {
 		for (hit = 1; hit <= targetHits; hit++) {
 			if (damage.includes(false)) break;
 			if (hit > 1 && pokemon.status === 'slp' && !isSleepUsable) break;
-			if (targets.some(target => target && !target.hp)) break;
+			if (targets.every(target => target && !target.hp)) break;
 			move.hit = hit;
 			if (move.smartTarget && targets.length > hit - 1) {
```

Comment:
Oh, this is ugly and not good.

Reply:
I guess this used to work because it used to be the case that at least one of `targetHits` and `targets.length` was 1, so that it made no difference whether you used `some` or `every`.

Reply:
Oh, no, I was specifically referring to `targets.length > hit - 1`, which I just fixed in 1d09dd11748176 but was confused how it worked in the past.

Reply:
Oh, are you talking about the `&&` to `||` change? I did wonder about that...